.evergreen/serverless/create-instance.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -o errexit
+set +o xtrace # disable xtrace to ensure credentials aren't leaked
+
+if [ -z "$PROJECT" ]; then
+    echo "Project name must be provided via PROJECT environment variable"
+    exit 1
+fi
+INSTANCE_NAME="$RANDOM-$PROJECT"
+
+if [ -z "$SERVERLESS_DRIVERS_GROUP" ]; then
+    echo "Drivers Atlas group must be provided via SERVERLESS_DRIVERS_GROUP environment variable"
+    exit 1
+fi
+
+if [ -z "$SERVERLESS_API_PRIVATE_KEY" ]; then
+    echo "Atlas API private key must be provided via SERVERLESS_API_PRIVATE_KEY environment variable"
+    exit 1
+fi
+
+if [ -z "$SERVERLESS_API_PUBLIC_KEY" ]; then
+    echo "Atlas API public key must be provided via SERVERLESS_API_PUBLIC_KEY environment variable"
+    exit 1
+fi
+
+echo "creating new serverless instance \"${INSTANCE_NAME}\"..."
+
+DIR=$(dirname $0)
+API_BASE_URL="https://account-dev.mongodb.com/api/atlas/v1.0/groups/$SERVERLESS_DRIVERS_GROUP"
+
+curl \
+  -u "$SERVERLESS_API_PUBLIC_KEY:$SERVERLESS_API_PRIVATE_KEY" \
+  --silent \
+  --show-error \
```

Comment:
Is there a risk that `--show-error` could leak the private key?

Reply:
It's certainly possible, though per [EVG-9754](https://jira.mongodb.org/browse/EVG-9754), all evergreen logs require authentication to be viewed, so the leakage would be limited to inside the company. That being said, we'd have to rotate the private key if it were leaked like that anyways, so I'm still not sure whether it's worth the risk or not to include error diagnostics. Happy to defer to your judgment on this.

Reply:
Ah true, this would only be in private logs. Then I'm not too worried, and I'm fine with keeping it.